tests/unit/test_dns_client.cpp

```diff
@@ -225,6 +225,8 @@ void InitTest(void)
 
 void FinalizeTest(void)
 {
+    AdvanceTime(30 * 1000);
+
```

Comment:
Not sure why this is needed. Is this intentional?

Reply:
Yes, this is intentional. We check if all allocated message buffers are freed at the end of the test. In this test, sometimes the test finishes quickly when there are still items in the delayed sender queue.

I did not change this in this pull request or the one that added `DelayedSender` as I wanted to keep the behavior the same as before, but I will probably add a new mechanism later to clear the `DelayedSender` upon `Mle::Stop()`. Existing code (which aligns with what we did before) sends all delayed messages in the queue regardless of current state.